Remotion/Core/UnitTests/Utilities/DateOnlyConverterTest.cs

```diff
@@ -0,0 +1,168 @@
+    var converter = new DateOnlyConverter();
+
+    var result = converter.ConvertFrom(Mock.Of<ITypeDescriptorContext>(), CultureInfo.CurrentCulture, new DateTime(1, 3, 4));
+
+    Assert.That(result, Is.InstanceOf<DateOnly>());
+    Assert.That(result, Is.EqualTo(new DateOnly(1, 3, 4)));
+  }
+
+  [Test]
+  public void ConvertFrom_NullableDateTime_WithValue ()
+  {
+    var converter = new DateOnlyConverter();
+
+    DateTime? dateTime = new DateTime(1, 3, 4);
+    var result = converter.ConvertFrom(Mock.Of<ITypeDescriptorContext>(), CultureInfo.CurrentCulture, dateTime);
+
+    Assert.That(result, Is.InstanceOf<DateOnly>());
+    Assert.That(result, Is.EqualTo(new DateOnly(1, 3, 4)));
+  }
+
+  [Test]
+  public void ConvertFrom_NullableDateTime_WithNull_ReturnsNull ()
+  {
+    var converter = new DateOnlyConverter();
+
+    DateTime? dateTime = null;
+    var result = converter.ConvertFrom(Mock.Of<ITypeDescriptorContext>(), CultureInfo.CurrentCulture, dateTime);
+
+    Assert.That(result, Is.EqualTo(null));
+  }
+
+  [Test]
+  public void ConvertFrom_WithUnsupportedInputValue_ThrowsNotSupportedException ()
+  {
+    var converter = new DateOnlyConverter();
+
+    Assert.That(
+        () => converter.ConvertFrom(Mock.Of<ITypeDescriptorContext>(), CultureInfo.CurrentCulture, "NotDateOnly"),
+        Throws.InstanceOf<NotSupportedException>()
+            .With.Message.EqualTo($"Cannot convert value from type '{typeof(string)}' to type '{typeof(DateOnly)}' or '{typeof(DateTime)}'."));
+  }
+
+  [Test]
+  public void ConvertTo_WithNullableDateOnlyInput ()
+  {
+    var converter = new DateOnlyConverter();
+
+    DateOnly? dateOnly = new DateOnly(1, 3, 4);
+    var result = converter.ConvertTo(Mock.Of<ITypeDescriptorContext>(), CultureInfo.CurrentCulture, dateOnly, typeof(DateTime));
+
+    Assert.That(result, Is.InstanceOf<DateTime>());
+    Assert.That(result, Is.EqualTo(new DateTime(1, 3, 4)));
+  }
+
+  [Test]
+  public void ConvertTo_WithNullInput_ReturnsNull ()
+  {
+    var converter = new DateOnlyConverter();
+
+    DateOnly? dateOnly = null;
+    var result = converter.ConvertTo(Mock.Of<ITypeDescriptorContext>(), CultureInfo.CurrentCulture, dateOnly, typeof(DateTime));
+
+    Assert.That(result, Is.Null);
+  }
+
+  [Test]
+  public void ConvertTo_DateTime ()
+  {
+    var converter = new DateOnlyConverter();
+
+    var result = converter.ConvertTo(Mock.Of<ITypeDescriptorContext>(), CultureInfo.CurrentCulture, new DateOnly(1, 3, 4), typeof(DateTime));
+
+    Assert.That(result, Is.InstanceOf<DateTime>());
+    Assert.That(result, Is.EqualTo(new DateTime(1, 3, 4)));
+  }
+
+  [Test]
+  public void ConvertTo_NullableDateTime ()
+  {
+    var converter = new DateOnlyConverter();
+
+    var result = converter.ConvertTo(Mock.Of<ITypeDescriptorContext>(), CultureInfo.CurrentCulture, new DateOnly(1, 3, 4), typeof(DateTime?));
+
+    Assert.That(result, Is.InstanceOf<DateTime?>());
+    Assert.That(result, Is.EqualTo(new DateTime(1, 3, 4)));
+  }
```

Comment:
at least the first test seems redundant with the above tests. I would suggest the following naming structure for all the ConvertTo-Tests:
WithDateOnlyInput_AndDateTimeDestinationType_ReturnsDateTime
WithNullableDateOnlyInput_AndNullableDateTimeDestinationType_ReturnsNullableDateTime
WithNullInput_AndNullableDateTimeDestinationType_ReturnsNull
etc